src/btree/bt_split.c

```diff
@@ -1413,8 +1413,9 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
      * was a forced eviction, in which case we leave the new page with the read generation unset.
      * Eviction will set the read generation next time it visits this page.
      */
-    if (!WT_READGEN_EVICT_SOON(orig->read_gen))
-        page->read_gen = orig->read_gen;
+    WT_READ_ONCE(orig_read_gen, orig->read_gen);
```

Comment:
This is kinda ugly, I know we discussed previously but as we're doing functions anyway I think just return the readgen from the function no?